### PR TITLE
feat(common): introduce SerialArrayBuilder and SerialArray

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Changes Made

- Implemented `HashKeySerDe` for `Serial`
- Added support for `Serial` to various functions and modules in the codebase
- Defined `SerialArrayBuilder` and `SerialArray` types for building and representing arrays of `Serial`
- Modified tests to reflect these changes

## Purpose

The purpose of these changes is to add support for a new `SERIAL` type and enable its usage in various functions and modules in the codebase. This is done by defining `SerialArrayBuilder` and `SerialArray` types for building and representing arrays of `Serial`. The changes also include updating tests to reflect these new additions.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by st1page — APPROVED** [review]
LGTM

File: src/common/src/array/serial_array.rs [hunk 1]
```diff
@@ -84,3 +92,68 @@ impl<'a> ScalarRef<'a> for Serial {
         self.0.hash(state)
     }
 }
+
+#[derive(Debug)]
+pub struct SerialArrayBuilder(PrimitiveArrayBuilder<i64>);
+
+#[derive(Debug, Clone)]
+pub struct SerialArray(PrimitiveArray<i64>);
```
Comment (xiangjinwu) [inline src/common/src/array/serial_array.rs:100]: What about `impl PrimitiveArrayItemType for Serial` and then `pub type SerialArray = PrimitiveArray<Serial>;` (see `NaiveDateArray`)?

Comment (shanicky) [reply src/common/src/array/serial_array.rs:100]: Cool, i'll try it.

**Review by xiangjinwu — APPROVED** [review]
lgtm

**Pull request merged**